src/docfx/lib/YamlUtility.cs

```diff
@@ -151,25 +163,33 @@ private static JToken ToJson(YamlNode node)
                 {
                     if (key is YamlScalarNode scalarKey)
                     {
-                        obj[scalarKey.Value] = ToJson(value);
+                        var jToken = ToJson(value, mappings);
+                        obj[scalarKey.Value] = jToken;
                     }
                     else
                     {
                         throw new NotSupportedException($"Not Supported: {key} is not a primitive type");
                     }
                 }
+                SetMappings(mappings, node, obj);
                 return obj;
             }
             if (node is YamlSequenceNode seq)
             {
                 var arr = new JArray();
                 foreach (var item in seq)
                 {
-                    arr.Add(ToJson(item));
+                    arr.Add(ToJson(item, mappings));
                 }
+                SetMappings(mappings, node, arr);
                 return arr;
             }
             throw new NotSupportedException($"Unknown yaml node type {node.GetType()}");
         }
+
+        private static void SetMappings(JTokenSourceMap mappings, YamlNode scalar, JToken value)
```

Comment:
you can make this method returns `JToken`, so the above return statements can remain single line:

```csharp
return SetMapping(mappings, scalar, new JValue(d))
```